### PR TITLE
[TwigBridge] Fix BC break due required twig environment

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes/no
| Fixed tickets | https://github.com/symfony/symfony/pull/24236#discussion_r149241720
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

See https://github.com/ezsystems/ezplatform-design-engine/pull/12 + https://github.com/silexphp/Silex-WebProfiler/pull/126 + https://github.com/bolt/bolt/pull/7154

Sorry for that :)

cc @fabpot

### CODE REVIEW

File: src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ class TwigDataCollector extends DataCollector implements LateDataCollectorInterf
     private $twig;
     private $computed;
 
-    public function __construct(Profile $profile, Environment $twig)
+    public function __construct(Profile $profile, Environment $twig = null)
```
Comment (derrabus) [inline src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php]: Maybe passing `null` should trigger a deprecation warning, so we can remove the boilerplate below on the 4.0 branch again.

Comment (ro0NL) [reply src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php]: sounds reasonable :) lets see what others think.

File: src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php [hunk 2]
```diff
@@ -62,24 +62,26 @@ public function lateCollect()
         $this->data['profile'] = serialize($this->profile);
         $this->data['template_paths'] = array();
 
-        $templateFinder = function (Profile $profile) use (&$templateFinder) {
-            if ($profile->isTemplate()) {
-                try {
-                    $template = $this->twig->load($name = $profile->getName());
-                } catch (\Twig_Error_Loader $e) {
-                    $template = null;
+        if (null !== $this->twig) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php]: `if (null === $this->twig) {return;}`?
about the deprecation, too late IMHO if we want to be strict about the feat freeze

**Review by stof — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

File: src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php]: No need to "deprecate" that as we are talking about an additional feature.

**Comment by GawainLynch** [comment]
Much appreciated @ro0NL :+1:

**Pull request closed without merging**